### DISCUSSION TITLE
Problem in Snap of components ie. Shift+O

### DISCUSSION THREAD

**Discussion by aryasubh** [SiEPIC/SiEPIC-Tools#164] [General]
I am facing the problem of snapping the components in KLayout using the command Shift+O. 
![1_snap_Issue](https://user-images.githubusercontent.com/41465929/145004549-d0fbfbd8-9f41-4365-865c-d8ee421dace6.JPG)
I further explore then this suggestion has been given
![2_No_transision_Suggestion](https://user-images.githubusercontent.com/41465929/145004637-32232528-822f-4519-afe7-0ec8c600d003.JPG)
I followed the suggested step as shown in this image
![3_transition_check](https://user-images.githubusercontent.com/41465929/145004705-351440e2-e3dc-468f-8cc7-e176fc3f0432.JPG)
I am using the 0.26.12 version of KLayout and SIEPIC 0.3.85 Py3 version 
Please help me to solve the problem.

**Comment by lukasc-ubc** [comment]
Hi Subhash,

It looks like you have might have two installations of SiEPIC-Tools (since it shows that there are two commands associated with Shift-O).  I suggest going in your KLayout folder and removing all the SiEPIC-Tools folders and installing again.

Also, we are only supporting KLayout 0.27 or greater.  Can you please install 0.27.5?

thank you
Lukas

**Reply by aryasubh** [reply]
Thank you so much Prof. Lukas for your prompt response. I shall do as sugguested and get back in case of any problem.

**Comment by aryasubh** [comment]
Dear Prof. Lukas,
I  installed KLayout 0.27.5 and then in the manage packages section, there were errors while installing SIEPIC tool and SIEPIC pdk. Thereafter, I installed a lower version of KLayout and was able to install SIEPIC tool and pdk, however, when I am trying to snap two components, it shows the error of " 'Layout' object has no attribute 'technology' " and in detailed section shows, " 'Layout' object has no attribute 'technology'
  C:\Users\aryas\KLayout\salt\siepic_tools\python\SiEPIC\extend.py:688
  C:\Users\aryas\KLayout\salt\siepic_tools\python\SiEPIC\extend.py:1660
  C:\Users\aryas\KLayout\salt\siepic_tools\python\SiEPIC\scripts.py:1260
  C:/Users/aryas/KLayout/salt/siepic_tools/pymacros/Keybindings/Layout/Snap Components.lym:7 "
Please help in this regard.  Earlier this issue was not there, since I have completed an edX course on Silicon Photonics and have Designed Photonic Circuits successfully. 
![Snap_problem](https://user-images.githubusercontent.com/41465929/147447856-1cdf30c6-9874-4a6e-abf1-5173246de849.JPG)

**Reply by mustafacc** [reply]
Hi Subhash,
Snapping will only work on the newer version of KLayout - if you're having trouble installing SiEPIC Tools through the package manager, i recommend checking the manual installation method:
https://www.youtube.com/watch?v=Y5a9kZVgZns

**Reply by aryasubh** [reply]
Dear Mustafa,
I followed your suggestion to install the latest version of SiEPIC tools manually as per your video demonstration, however, I am facing the error, "invalid Switch ". How to overcome it? Here is the snapshot of the problem. 
![SiEPIC tool manual installation error](https://user-images.githubusercontent.com/41465929/149209108-1d488902-d55e-4661-9558-d45510f5bdb4.JPG)

**Comment by aryasubh** [comment]
Dear Mustafa,
Thank you for your answer, however, I could solve the issue in the following way.
1. I install an older version of KLayout and installed SIEPIC tool and pdk.
2. Thereafter, I uninstall an older version of KLayout i.e. 0.26.12 from the control panel and did not delete the Klayout file in C drive.
3. And then, I installed the newest version of KLayout i.e. 0.27.5, observed that SIEPIC tools and pdk were already there.
4. Lastly, I designed MZI using the Shift+O command for snapping components.

**Comment by aryasubh** [comment]
Hello Prof Lukas and team,
Happy new Year-2022! 
I am again facing a problem while updating SIEPIC tool 0.3.88.
![SIEPIC tool installation error](https://user-images.githubusercontent.com/41465929/149086556-eb656c12-6a7d-44bf-ad14-65cf08b9720c.JPG)
While doing so, I lost previous version of SIEPIC tool and hence, not able to use its library components. 
Please suggest solving this issue. 
Thank you so much.

**Comment by lukasc-ubc** [comment]
Hi Subhash,

We have been seeing intermittent problems with the Salt package manager, where sometimes large files have difficulty downloading.  That file hasn’t been changed (for over a year), so I don’t think the latest release changes should affect the installation.

A suggestion would be to install SiEPIC tools using GitHub Desktop, and create a symbolic link from the KLayout folder to the GitHub folder.  This is the way I have it installed, and makes it easy to update.

https://github.com/SiEPIC/SiEPIC-Tools/wiki/Contributing
And there is a script to create the symbolic links for you: https://github.com/SiEPIC/SiEPIC-Tools/tree/master/Documentation/Developers

**Reply by aryasubh** [reply]
Thank you Prof. Lukas for responding and providing valuable suggestions. Let me try it. 
With regards,
(Subhash Arya)

**Comment by lukasc-ubc** [comment]
maybe /D instead of D/?

**Reply by aryasubh** [reply]
Thank you so much for correcting the mistake. It is now working. I am able to install the latest version of SiEPIC tools. Thank you, Prof. Lukas and Mustafa. 
![SiEPIC tool manual installation error_removed](https://user-images.githubusercontent.com/41465929/149210832-7aed995f-7837-4403-9269-7ca7ec85144a.JPG)
![SiEPIC latest version](https://user-images.githubusercontent.com/41465929/149210852-29ae6c28-c863-4150-b61c-84d77cabc38b.JPG)
